### PR TITLE
Plugin 2.0

### PR DESCRIPTION
**Pull request by malkafly-cw**
There is some //TODO on code while I'm working on PIX changes/fixes. Please don't approve before I send a new commit removing these //TODO. Thanks!

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -3,5 +3,10 @@
     "description": "Offer to your clients the best interest in e-Commerce by using InfinitePay as your payment method.",
     "require": {
         "php": ">=5.6"
+    },
+    "autoload": {
+        "psr-4": {
+            "Woocommerce\\InfinitePay\\": "src/"
+        }
```
Comment (panachi-cw) [inline composer.json:10]: This is applied only for development environment or Wordpress also uses what is described here somehow?

File: i18n/languages/infinitepix-woocommerce-pt_BR.po [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+"Language: pt_BR\n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+"Plural-Forms: nplurals=2; plural=(n > 1);\n"
+"X-Generator: Poedit 3.1\n"
+"X-Domain: infinitepay-woocommerce\n"
+
+#. Plugin Name of the plugin
+msgid "InfinitePay payments for WooCommerce"
```
Comment (panachi-cw) [inline i18n/languages/infinitepix-woocommerce-pt_BR.po:21]: `InfinitePay for WooCommerce`

File: src/Controller/Credentials.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+// 	curl --request POST \
+//   --url https://api.infinitepay.io/v2/oauth/token \
+//   --header 'Content-Type: application/json' \
+//   --data '{
+//         "grant_type": "client_credentials",
+//         "client_id": "[client_id]",
+//         "client_secret": "[client_secret]",
+//         "scope": "[access_token_scope]"
+//     }'
```
Comment (panachi-cw) [inline src/Controller/Credentials.php:28]: ✂️

File: src/Helper/ApiInfinitePay.php [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+                $endpoint = $this->environment == 'sandbox' ? 'https://authorizer-staging.infinitepay.io/v2/transactions' : 'https://api.infinitepay.io/v2/transactions';
+                $this->args['headers']['Authorization'] = 'Bearer ' . $bodyAuth['access_token'];
+                $this->args['body'] = json_encode($body, JSON_UNESCAPED_UNICODE);
+
+                //TODO: Remover estas linhas de log do PR
+                //TODO: Remover estas linhas de log do PR
+                //TODO: Remover estas linhas de log do PR
+                //TODO: Remover estas linhas de log do PR
+                //TODO: Remover estas linhas de log do PR
+                //TODO: Remover estas linhas de log do PR
```
Comment (panachi-cw) [inline src/Helper/ApiInfinitePay.php:112]: Don't forget to ✂️

File: src/Helper/Utils.php [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+
+    // public static function log( $data, $log_name = 'debug' )
+	// {
+	// 	$name = sprintf( '%s-%s.log', $log_name, date( 'd-m-Y' ) );
+	// 	$log  = print_r( $data, true ) . PHP_EOL;
+	// 	$log .= "\n=============================\n";
+
+	// 	file_put_contents( Core::get_file_path( $name, 'logs/' ), $log, FILE_APPEND );
+	// }
```
Comment (panachi-cw) [inline src/Helper/Utils.php:102]: ✂️

File: src/InfinitePayCore.php [hunk 6]
```diff
@@ -0,0 +1,385 @@
...
+		?>
+		<h2>InfinitePay</h2>
+		<?php if (!$this->api->has_access_token): ?>
+			<div id="message" class="notice-warning notice">
+			<?php
+			printf(
+					'<p><strong>%s: </strong>%s <strong>%s</strong>.</p>',
+					__('Before accepting payments', 'infinitepay-woocommerce'),
+					__('you need to add your', 'infinitepay-woocommerce'),
+					__('Client ID and Client Secret', 'infinitepay-woocommerce')
```
Comment (panachi-cw) [inline src/InfinitePayCore.php:82]: Please add `Access the <a href="https://money.infinitepay.io/settings/credentials">InfinitePay dashboard</a> to obtain your credentials` on this message

File: readme.txt [hunk 7]
```diff
@@ -49,9 +49,9 @@ Done! It will be in the "Installed Plugins" section and from there you can activ
 If you installed it correctly, you will see it in your list of "Installed Plugins" on the WordPress work area. Please enable it and input your api key on the specified field.
 
 == Changelog ==
-- 1.1.10 (2022/08/10) -
-* Bug fixes
-    - Fix cache setting for unsupported PHP versions
+- 2.0.0 (2022/08/08) -
+* New version and Bug fixes
```
Comment (panachi-cw) [inline readme.txt:53]:
```
- 2.0.0 (2022/08/26)
* Accepts PIX payments
* Using to a more secure authentication protocol
* Added mechanism to configure order status from-to payment status
* A lot of bug fixes and enhancements
* Enjoy :)

**Review by panachi-cw — CHANGES_REQUESTED** [review]
Please use `InfinitePay for WooCommerce` for all plugin naming/title references

File: composer.json (hunk 1 above)
Comment (malkafly-cw) [reply composer.json:10]: It's need to the plugin working with namespace as the new version does

Comment (malkafly-cw) [reply composer.json:10]: for any environment.

File: src/Controller/Credentials.php (hunk 3 above)
Comment (malkafly-cw) [reply src/Controller/Credentials.php:28]: all these comments will be removed on the final commit, but thanks for the warning o/

File: src/InfinitePayCore.php (hunk 6 above)
Comment (malkafly-cw) [reply src/InfinitePayCore.php:82]: Okey!

File: readme.txt [hunk 8]
```diff
@@ -49,9 +49,12 @@ Done! It will be in the "Installed Plugins" section and from there you can activ
 If you installed it correctly, you will see it in your list of "Installed Plugins" on the WordPress work area. Please enable it and input your api key on the specified field.
 
 == Changelog ==
-- 1.1.10 (2022/08/10) -
-* Bug fixes
-    - Fix cache setting for unsupported PHP versions
+- 2.0.0 (2022/08/26) -
+* Accepts PIX payments
+* Using to a more secure authentication protocol
```
Comment (panachi-cw) [inline readme.txt:54]:
```suggestion
* Using a more secure authentication protocol
```

File: src/Helper/Log.php [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+		}
+		return $this->log;
+	}
+
+	public function write_log( $function, $message ) {
+		$options = get_option('woocommerce_infinitepay_settings');
+
+		//if (issrt($options['enabled_log']) && $options['enabled_log'] == 'yes') {
+			$this->log->add( $this->id, '[' . $function . ']: ' . $message );
+		//}
```
Comment (panachi-cw) [inline src/Helper/Log.php:38]: This is right?

File: woocommerce-infinitepay.php [hunk 10]
```diff
@@ -1,52 +1,55 @@
 <?php
 /*
- * Plugin Name: InfinitePay for WooCommerce
+ * Plugin Name: InfinitePay payments for WooCommerce
```
Comment (panachi-cw) [inline woocommerce-infinitepay.php:3]:
```suggestion
 * Plugin Name: InfinitePay for WooCommerce
```

**Review by panachi-cw — CHANGES_REQUESTED** [review]

File: src/Helper/Log.php (hunk 9 above)
Comment (malkafly-cw) [reply src/Helper/Log.php:38]: Fixed, the previous plugin version don't respect the enabled choice

File: src/index.js [hunk 11]
```diff
@@ -103,135 +32,103 @@
...
+            clearErrors();
+            if(document.getElementById('ip_method').value == 'cc-form') {
+                
+                var form = document.querySelector('form.woocommerce-checkout');
+                if(!form) {
+                    form = document.querySelector('form#order_review');
+                }
+                var ipay = new IPay({ access_token: wc_infinitepay_params.access_token });
+
+                ipay.listeners = {
```
[the remark this answers is not in the document]
Comment (malkafly-cw) [reply src/index.js:83]: nom nom nomm

**Review by bru-carenzi — APPROVED** [review]
### Test Summary

**Status**: 👍 Ready 

---


📋 Test report:
[Case tests](https://docs.google.com/spreadsheets/d/1dvzYJpYxHxr_10zEEMP3U9YMfj1G9Xe_cANN1xCqde0/edit#gid=323694731)

 ❗ **Blocking issues**
- Nop

**Pull request merged**